crypto/pem/pem_test.cc

```diff
@@ -43,3 +58,141 @@ TEST(PEMTest, NoRC4) {
   EXPECT_EQ(ERR_LIB_PEM, ERR_GET_LIB(err));
   EXPECT_EQ(PEM_R_UNSUPPORTED_ENCRYPTION, ERR_GET_REASON(err));
 }
+
+struct TmpFileDeleter {
+  void operator()(FILE *f) const { fclose(f); }
+};
+
+using TmpFilePtr = std::unique_ptr<FILE, TmpFileDeleter>;
+
+static void* d2i_ASN1_INTEGER_void(void ** out, const unsigned char **inp, long len) {
+  return d2i_ASN1_INTEGER((ASN1_INTEGER **)out, inp, len);
+}
+
+static int i2d_ASN1_INTEGER_void(const void *a, unsigned char **out) {
+  return i2d_ASN1_INTEGER((ASN1_INTEGER *)a, out);
+}
+
+TEST(PEMTest, WriteReadASN1IntegerPem) {
+#if defined(OPENSSL_ANDROID)
+  // On Android, when running from an APK, |tmpfile| does not work. See
+  // b/36991167#comment8.
+  GTEST_SKIP();
+#endif
+  // Numbers for testing
+  long nums[8] = {
+      0x00000001L,
+      0x00000100L,
+      0x00010000L,
+      0x01000000L,
+      -2L};
+
+  for(size_t i = 0; i < 8; i++) {
+    long original_value = nums[i];
+
+    // Create an ASN1_INTEGER with value 42
```

Comment:
* The number of elements in `nums` seems to be less than `8` in this test. 
* We may want to adjust this comment here?